Dsse one sig

kommendorkapten opened the pull request:
#### Summary
Closes #225 

This is implemented by verifying that the envelope has _exactly_ one signature during the verification process. This is a bug-fix, as the protobuf spec for the bundle requires exactly one signature to be present in the envelope.

Note that I did not chose to modify the `Signature` [function](https://github.com/sigstore/sigstore-go/blob/main/pkg/bundle/signature_content.go#L100). This could be done too, but I think it is better to have the verification process be more strict, as it is a security feature.

#### Release Note
* Signature verification of bundles with DSSE envelopes now verifies that the envelope have exactly one signature.

#### Documentation
N/A

kommendorkapten committed:
Ran go fmt

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>

kommendorkapten committed:
Added test to verify that dsse envelope has exact one signature.

This is a bug-fix, as the protobuf spec for the bundl requires exactly one
signature to be present in the envelope.

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>

kommendorkapten committed:
Spelling error.

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>

kommendorkapten committed:
Correct year in copyright notice.

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>

kommendorkapten committed:
Lint fixes: removed unnecessary newline.

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>

codysoyland commented on pkg/verify/signature.go:
```diff
@@ -31,6 +31,8 @@ import (
 	"github.com/sigstore/sigstore/pkg/signature/options"
 )
 
+var ErrInvSigCount = errors.New("exactly one signature is required")
```
For consistency with other sentinel errors which use unabbreviated names, could you rename this `ErrInvalidSignatureCount`?

Hayden-IO replied:
Can we also make this specific to DSSE as well, mentioning DSSE in the var name and error message?

kommendorkapten replied:
Makes sense. Updated.

codysoyland commented on pkg/verify/dsse_test.go:
```diff
@@ -0,0 +1,124 @@
+	return e.e
+}
+
+func TestVerifyEnvelope(t *testing.T) {
```
I would like to be explicit what the purpose of the test is, for example (or add a comment if you prefer):
```suggestion
func TestVerifyEnvelopeSignatureCount(t *testing.T) {
```

kommendorkapten replied:
Yeah, good call out. Name is too generic.

kommendorkapten committed:
Updates from feeback.

Better error name, more explicit about the usage.
More explicit test caste name.

Signed-off-by: Fredrik Skogman <kommendorkapten@github.com>
